nilearn/plotting/__init__.py

```diff
@@ -3,26 +3,31 @@
 """
 # Authors: Chris Filo Gorgolewski, Gael Varoquaux
 
+from ..version import (_import_module_with_version_check,
+                       OPTIONAL_MATPLOTLIB_MIN_VERSION)
+
 ###############################################################################
 # Make sure that we don't get DISPLAY problems when running without X on
 # unices
 
-def _set_mpl_backend():
-    # We are doing local imports here to avoid poluting our namespace
-    import os
-    if not os.name == 'posix':
-        # We have the problem only on posix systems
-        return
-    if 'DISPLAY' in os.environ:
-        return
 
+def _set_mpl_backend():
     try:
+        # We are doing local imports here to avoid poluting our namespace
         import matplotlib
-        # Agg is a backend that will do PNGs and PDFs
-        matplotlib.use('Agg')
+        import os
+        # We have the problem only on posix systems
+        if os.name == 'posix' and 'DISPLAY' not in os.environ:
+            # Agg is a backend that will do PNGs and PDFs
+            matplotlib.use('Agg')
     except ImportError:
-        pass
         # No need to fail here, eg during tests
+        pass
+    else:
+        # When matplotlib was successfully imported we need to check
+        # that the version is greater that the minimum required one
+        _import_module_with_version_check('matplotlib',
+                                          OPTIONAL_MATPLOTLIB_MIN_VERSION)
```

Comment:
I had to shuffle things around a little bit here. Basically I both wanted
* the matplotlib minimum version should be always checked 
* Agg backend should only be used for posix platforms when the DISPLAY environment variable is not defined (e.g. on Travis)